src/claimtrie.cpp

```diff
@@ -1623,14 +1610,14 @@ bool CClaimTrieCache::removeClaimFromQueue(const std::string& name, const COutPo
 bool CClaimTrieCache::undoAddClaim(const std::string& name, const COutPoint& outPoint, int nHeight) const
 {
     int throwaway;
-    const std::string newName = base->normalizeClaimName(name);
+    const std::string newName = normalizeClaimName(name);
     return removeClaim(newName, outPoint, nHeight, throwaway, false);
 }
 
-bool CClaimTrieCache::spendClaim(const std::string& name, const COutPoint& outPoint, int nHeight, int& nValidAtHeight) const
+bool CClaimTrieCache::spendClaim(std::string& name, const COutPoint& outPoint, int nHeight, int& nValidAtHeight) const
```

Comment:
The out parameter is used in mining.cpp and main.cpp (to avoid duplicate work that name). Out parameter usage is not very obvious, though. I'm open to other suggestions.

Reply:
Yeah, i see now. The right approach, to me, is to call normalize before spend, we can prevent double work if we have function like isNormlized before real normalization. But i don't think double call will slow down significantly.